Set `logs_config.cca_in_ad: true` by default

djmitche opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

Enables a feature flag by default.

### Motivation

This feature has not changed since 7.39, where we have been incrementally enabling it and will continue to do so.   Assuming no further issues are found during 7.39 testing, nor during 7.40 QA, this will enable the flag by default for all customers in the 7.40 release.

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

7.39 QA of the relevant functionality set this flag to true, so no additional functionality-specific QA is required.  Other agent QA may uncover unanticipated consequences of this change.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

djmitche committed:
Set `logs_config.cca_in_ad: true` by default

cswatt commented on releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml:
```diff
@@ -0,0 +1,14 @@
+---
+other:
+  - |
+    The ``logs_config.cca_in_ad`` feature flag now defaults to true.  This
```
is this supposed to have double backticks?

djmitche replied:
Yes, it's reStructuredText.

cswatt commented on releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml:
```diff
@@ -0,0 +1,14 @@
+other:
+  - |
+    The ``logs_config.cca_in_ad`` feature flag now defaults to true.  This
+    selects updated codepaths in autodiscovery and the logs-agent.  No behavior
```
```suggestion
    selects updated codepaths in Autodiscovery and the Logs Agent.  No behavior
```

djmitche committed:
Update releasenotes/notes/cca_in_ad_on-5ec511f4f574337b.yaml

Co-authored-by: cecilia saixue watt <cecilia.watt@datadoghq.com>

djmitche commented:
Oh, I bet those test failures _are_ related, as the defaults cause random fluctuations in expected values :(

djmitche committed:
Look at actually-scheduled configs in tests

djmitche committed:
Remove autoconfig.TestOverride

Overrides are now handled by the specific service implementations to
which they relate, and are tested there.

djmitche committed:
Remove tests for service configs

With cca_in_ad, these are no longer a thing.
